src/python/py/models/builder.py

```diff
@@ -1338,11 +1349,10 @@ def make_attention_mask_reformatting_for_mha(self):
         end_add_shape = ["batch_size", 1, "source_sequence_length", "target_sequence_length"]
         self.make_add(end_add_name, end_add_inputs, dtype=self.io_dtype, shape=end_add_shape) # Shape of mask is now (B, 1, S, T)
 
-        # TODO: replace Concat with Expand for performance gains
         concat_name = f"{basename}/Concat"
-        concat_inputs = [f"{end_add_name}/output_0" for _ in range(self.num_attn_heads)]
+        concat_inputs = [f"{end_add_name}/output_0", f"/model/constants/TensorProto.INT64/1D/1, {self.num_kv_heads}, 1, 1"]
```

Comment:
ort-web/webgpu has a limit on the number of buffers and a concat with 32 inputs isn't going to work. 
Replace it with Tile.
It is enabled for all flavors since term should be better. But if we have concerns about it I can special case it for ort-web.

Reply:
Let me check how tile works on cpu and cuda. and not sure if dml supports Tile operator

Reply:
ah, after reading it carefully, Tile is much better.